packages/web/src/common/store/pages/audio-rewards/slice.ts

```diff
@@ -97,18 +97,13 @@ const slice = createSlice({
       if (userChallenges === null) {
         state.userChallenges = {}
       } else {
-        state.userChallenges = userChallenges.reduce((acc, challenge) => {
-          // TODO: AUD-1536
-          // Remove this custom override logic when we have identified the proper fix
-          if (
-            challenge.challenge_id === 'listen-streak' &&
-            challenge.is_complete
-          ) {
-            challenge.current_step_count = challenge.max_steps
-          }
-          acc[challenge.challenge_id] = challenge
-          return acc
-        }, {} as Partial<Record<ChallengeRewardID, UserChallenge>>)
+        state.userChallenges = userChallenges.reduce(
+          (acc, challenge) => ({
+            ...acc,
+            [challenge.challenge_id]: challenge
+          }),
+          {} as Partial<Record<ChallengeRewardID, UserChallenge>>
```

Comment:
not something you introduced, but can we type the `acc` or the reduce<> with this object type so we can prevent a cast?